ospd_openvas/nvticache.py

```diff
@@ -91,13 +91,20 @@ def get_oids(self) -> Iterator[Tuple[str, str]]:
         Returns:
             An iterable of tuples of file name and oid.
         """
+        # hack for gvmd; we need to verify why gvmd is requiring a sorted list
+        # for hash verification. This does require a lot of memory and
+        # calculation time compared to a simple yield (f, oid)
+
+        result = []
         if self.notus:
             for f, oid in self.notus.get_filenames_and_oids():
-                yield (f, oid)
+                result.append((f, oid))
         if self.ctx:
             for f, oid in OpenvasDB.get_filenames_and_oids(self.ctx):
                 if not self.notus or not self.notus.exists(oid):
-                    yield (f, oid)
+                    result.append((f, oid))
+        result.sort()
```

Comment:
I think you should sort the list by the oid element of the tuples like this:
```suggestion
        result.sort(key=lambda t: t[1])
```
Also, gvmd only needs the VTs to be sorted for the hash calculation, so it may be a better idea to create a list of just the OIDs in the hash calculation function and then sort this list. That way the list items do not have to store the file names, saving some memory.

Reply:
right, my mistake it wasn't sorted by oid but by filename; for the second I remark as far as I can tell it is calling in this case `<get_vts filter='%s'/>` in the end so I don't see how it is limited by hash calculation?

Reply:
Ahh I think now I got it. The order doesn't matter while processing the actual NVT in gvmd at all since it does it per oid anyway.